status-session-management-spec.md

```diff
@@ -0,0 +1,118 @@
+
+## Table of Contents
+  - [Abstract](#abstract)
+  - [Introduction](#introduction)
+  - [Initialization](#initialization)
+  - [Concurrent sessions](#concurrent-sessions)
+  - [Re-keying](#re-keying)
+  - [Multi-device support](#multi-device-support)
+  - [Pairing](#pairing)
+  - [Sending messages to a paired group](#sending-messages-to-a-paired-group)
+  - [Account recovery](#account-recovery)
+  - [Partitioned devices](#partitioned-devices)
+  - [Trust establishment](#trust-establishment)
+    - [Contact request](#contact-request)
+  - [Expired session](#expired-session)
+  - [Stale devices](#stale-devices)
+
+## Introduction
+
+A peer is identified by two pieces of data:
+
+1) An `installation-id` which is generated upon creating a new account in the `Status` application
+2) Their identity whisper key
+
+## Initialization
+
+A new session is initialized once a successful X3DH exchange has taken place. Subsequent messages will use the established session until re-keying is necessary.
+
+## Concurrent sessions
+
+If two sessions are created concurrently between two peers the one with the symmetric key, first in byte order should be used this marks that the other has expired.
+
+## Re-keying
+
+On receiving a bundle from a given peer with a higher version, the old bundle should be marked as expired and a new session should be established on the next message sent.
+
+## Multi-device support
+
+Multi-device support is quite challenging as we don't have a central place where information on which and how many devices (identified by their respective `installation-id`) belongs to a whisper-identity.
+
+Furthermore we always need to take account recovery in consideration, where the whole device is wiped clean and all the information about any previous sessions is lost.
+
+Taking these considerations into account, the way multi-device information is propagated through the network is through bundles/contact codes, which will contain information about paired devices as well as information about the sending device.
+
+This mean that every time a new device is paired, the bundle needs to be updated and propagated with the new information, and the burden is put on the user to make sure the pairing is successful.
+
+The method is loosely based on https://signal.org/docs/specifications/sesame/ .
+
+<!-- TODO: This multi device section isn't clear enough -->
+<!-- TODO: Additionally, it seems tightly coupled with secure transport, which makes things like multi device public chats harder to reason about (IMO). E.g. as a client impl I might want multi device support but not want to impl double ratchet etc, so what does this mean? -->
+
+## Pairing
+
+When a user adds a new account in the `Status` application, a new `installation-id` will be generated. The device should be paired as soon as possible if other devices are present. Once paired the contacts will be notified of the new device and it will be included in further communications.
```

Comment:
Is this all part of session?

Reply:
Probably not, but I don't see where else we would add this pairing info. Accounting?